recipes/unix.rb

```diff
@@ -0,0 +1,190 @@
+  sdk_checksum = node['android-sdk']['checksum']['mac_os_x']
+else
+  Chef::Log.error('Platform not supported!')
+end
+
+ark node['android-sdk']['name'] do
+  url sdk_url
+  path node['android-sdk']['setup_root']
+  checksum sdk_checksum
+  version node['android-sdk']['version']
+  prefix_root node['android-sdk']['setup_root']
+  prefix_home node['android-sdk']['setup_root']
+  owner node['android-sdk']['owner']
+  group node['android-sdk']['group'] unless mac_os_x?
+  backup node['android-sdk']['backup_archive']
+  action node['android-sdk']['with_symlink'] ? :install : :put
+end
+
+#
+# Fix non-friendly 0750 permissions in order to make android-sdk available to all system users
+#
+%w(add-ons platforms tools).each do |subfolder|
+  directory File.join(android_home, subfolder) do
+    mode '0755'
+  end
+end
+# TODO: find a way to handle 'chmod stuff' below with own chef resource (idempotence stuff...)
+execute 'Grant all users to read android files' do
+  command "chmod -R a+r #{android_home}/*"
+  user node['android-sdk']['owner']
+  group node['android-sdk']['group'] unless mac_os_x?
+end
+execute 'Grant all users to execute android tools' do
+  command "chmod -R a+X #{File.join(android_home, 'tools')}/*"
+  user node['android-sdk']['owner']
+  group node['android-sdk']['group'] unless mac_os_x?
+end
+
+#
+# Configure environment variables (ANDROID_HOME and PATH)
+#
+if linux?
+  template "/etc/profile.d/#{node['android-sdk']['name']}.sh" do
+    source 'android-sdk.sh.erb'
+    mode '0644'
+    owner node['android-sdk']['owner']
+    group node['android-sdk']['group'] unless mac_os_x?
+    variables(
+      android_home: android_home
+    )
+    only_if { node['android-sdk']['set_environment_variables'] }
+  end
+elsif mac_os_x?
+  bash_profile 'profile.android_sdk' do
+    user node['android-sdk']['owner']
+    content <<-EOH
+      export ANDROID_HOME=#{android_home}
+      export PATH=$PATH:$ANDROID_HOME/tools:$ANDROID_HOME/platform-tools
+    EOH
+  end
+else
+  Chef::Log.error('Platform not supported!')
+end
+
+#
+# Install, Update (a.k.a. re-install) Android components
+#
+
+# KISS: use a basic idempotent guard, waiting for https://github.com/gildegoma/chef-android-sdk/issues/12
+unless File.exist?("#{setup_root}/#{node['android-sdk']['name']}/temp")
+
+  # With "--filter node['android-sdk']['components'].join(,)" pattern,
+  # some system-images were not installed as expected.
+  # The easiest way I could find to fix this problem consists
+  # in executing a dedicated 'android sdk update' command for each component to be installed.
+  node['android-sdk']['components'].each do |sdk_component|
+    script "Install Android SDK component #{sdk_component}" do
+      interpreter 'expect'
+      environment 'ANDROID_HOME' => android_home
+      path [File.join(android_home, 'tools')]
+      user node['android-sdk']['owner']
+      group node['android-sdk']['group'] unless mac_os_x?
```

Comment:
use http://www.rubydoc.info/github/opscode/chef/Chef/Mixin/HomebrewUser#find_homebrew_uid-instance_method :) this should solves your problem with `user`.